compiler/src/iree/compiler/Codegen/Common/DecomposeConvolutionToLowerDimOps.cpp

```diff
@@ -16,6 +19,110 @@ namespace mlir::iree_compiler {
 
 namespace {
 
+static bool foldHDim(linalg::DepthwiseConv2DNhwcHwcOp convOp) {
+  Value kernel = convOp.getInputs().back();
+  Value output = convOp.getOutputs().front();
+
+  auto kernelType = dyn_cast<RankedTensorType>(kernel.getType());
+  auto outputType = dyn_cast<RankedTensorType>(output.getType());
+
+  auto kernelShape = kernelType.getShape();
+  auto outputShape = outputType.getShape();
+
+  int64_t khSize = kernelShape[0];
+  int64_t ohSize = outputShape[1];
+  bool removeH = (khSize == 1 && ohSize == 1);
+
+  return removeH;
+}
+
+/// Computes a "decomposed" lowering config attribute for a conv OP
+///
+/// This method complements the patterns to decompose 2D convolutions into 1D
+/// convs. Specifically, it will update the lowering config attached to a Conv
+/// Op in a way that matches the "decomposition" patterns.
+///
+/// At the moment only Depthwise HWC convolutions are supported.
+static FailureOr<IREE::Codegen::LoweringConfigAttr>
+computeNewLCA(ArrayRef<Operation *> computeOps, MLIRContext *context) {
+
+  // 0.1 Double-check that there's only one convolution Op.
+  // TODO: Make this hook work with multiple conv Ops
+  assert(llvm::count_if(computeOps,
+                        [](Operation *op) {
+                          return isa<linalg::ConvolutionOpInterface>(op);
+                        }) == 1 &&
+         "Exactly 1 Linalg Conv Op is expected");
+
+  // 1. Get the conv Op to update
+  // 1.1. ATM only 2D depthwise HWC convs are supported.
+  // TODO: Add support for other convs
+  linalg::DepthwiseConv2DNhwcHwcOp convOp;
+  for (auto op : computeOps) {
+    if (isa<linalg::DepthwiseConv2DNhwcHwcOp>(op)) {
+      convOp = cast<linalg::DepthwiseConv2DNhwcHwcOp>(op);
+      break;
+    }
+  }
+
+  if (!convOp) {
+    return failure();
+  }
+
+  // 1.2. ATM only folding of the H dim is supported.
+  // TODO: Add support for cases where the W dim is folded.
+  if (!foldHDim(convOp))
+    return failure();
+
+  // 2. Get the current lowering config attached to the Conv Op.
+  FailureOr<IREE::Codegen::LoweringConfigAttr> loweringConfigAttr =
+      getLoweringConfig(computeOps);
+  if (failed(loweringConfigAttr))
+    return failure();
+
+  // TODO: Either remove "interchange" from lowering_config or add support in
+  // this pass.
+  if (!loweringConfigAttr->isInterchangeEmpty())
+    return failure();
+
+  // 3. Calculate new tiling levels.
+  // Note that this will basically erase the _H_ dims from the orignal lowering
+  // config.
+  auto dims = linalg::inferConvolutionDims(convOp);
+  SmallVector<unsigned> hDimsToErase{dims->outputImage[0], dims->filterLoop[0]};
```

Comment:
Please use assignment syntax. The brace initialization with constructor syntax is dangerous. It could potentially lead to a bug, e.g.,

```
// Could invoke an intializer list constructor, or a two-argument constructor.
Frobber frobber{size, &bazzer_to_duplicate};

// Makes a vector of two doubles.
std::vector<double> fifty_pies{50, 3.14};
```

For more context, see https://abseil.io/tips/88

```suggestion
  SmallVector<unsigned> hDimsToErase = {dims->outputImage[0], dims->filterLoop[0]};
```